examples/DLM/ex0/example.cpp

```diff
@@ -0,0 +1,789 @@
+                *fem_v_system.current_local_solution = *ibfe_v_system.current_local_solution;
+            }
+            for (int k = 0; k < num_fe_substeps; ++k)
+            {
+                const double sub_dt = 0.5 * dt / num_fe_substeps;
+                const double start_time = loop_time + k * sub_dt;
+                const double end_time = loop_time + (k + 1) * sub_dt;
+                fem_solver->preprocessIntegrateData(start_time,
+                                                    end_time,
+                                                    /*num_cycles*/ 1);
+                fem_solver->modifiedTrapezoidalStep(start_time, end_time);
+                fem_solver->postprocessIntegrateData(start_time,
+                                                     end_time,
+                                                     /*num_cycles*/ 1);
+            }
+
+            // Map FEM solver data to the IBFE method ops and perform a full time step with the IB solver.
+            {
+                System& fem_x_system =
+                    fem_equation_systems->get_system(FEMechanicsExplicitIntegrator::COORDS_SYSTEM_NAME);
+                *ibfe_y_system.solution = *fem_x_system.solution;
+                *ibfe_y_system.current_local_solution = *fem_x_system.current_local_solution;
+
+                System& fem_u_system =
+                    fem_equation_systems->get_system(FEMechanicsExplicitIntegrator::VELOCITY_SYSTEM_NAME);
+                *ibfe_v_system.solution = *fem_u_system.solution;
+                *ibfe_v_system.current_local_solution = *fem_v_system.current_local_solution;
+            }
+            time_integrator->advanceHierarchy(dt);
+
+            // Map IBFE data to the FEM solver and perform half a time step with the structural solver.
+            {
+                System& ibfe_x_system = ibfe_beam_equation_systems->get_system(IBFEMethod::COORDS_SYSTEM_NAME);
+                *fem_y_system.solution = *ibfe_x_system.solution;
+                *fem_y_system.current_local_solution = *ibfe_x_system.current_local_solution;
+
+                System& ibfe_u_system =
+                    fem_equation_systems->get_system(FEMechanicsExplicitIntegrator::VELOCITY_SYSTEM_NAME);
+                *fem_v_system.solution = *ibfe_u_system.solution;
+                *fem_v_system.current_local_solution = *ibfe_v_system.current_local_solution;
+            }
+            for (int k = 0; k < num_fe_substeps; ++k)
+            {
+                const double sub_dt = 0.5 * dt / num_fe_substeps;
+                const double start_time = loop_time + 0.5 * dt + k * sub_dt;
+                const double end_time = loop_time + 0.5 * dt + (k + 1) * sub_dt;
+                fem_solver->preprocessIntegrateData(start_time,
+                                                    end_time,
+                                                    /*num_cycles*/ 1);
+                fem_solver->modifiedTrapezoidalStep(start_time, end_time);
```

Comment:
Interesting approach. A major source of complication in `DLMMethod` (from the DLM project) was that we interpolated positions in time - it looks like that is skipped here.

Reply:
You don't need to do that if you couple via fractional stepping.

Reply:
I think fractional stepping (e.g. using Strang) makes a lot of sense here because the idea (at least for stiff elasticity) is that you want to be taking smaller structural time steps _anyway_.

Reply:
Is there any loss of accuracy from holding the IBFE position constant over the substeps?

Reply:
Surely it impacts accuracy --- but I think it does not impact order of accuracy. Also, I don't think this is the major determinant of accuracy here.

(BTW, we do the same thing here: https://arxiv.org/abs/2204.12907.)

Reply:
BTW, presumably the better way to do this is to use a more modern time stepping scheme that properly handles multiple time scales.

Reply:
I might give this a try with the deal.II code later :)

Reply:
I just added some input files demonstrating how to increase structural stiffness without reducing the overall time step size (although, of course, it is necessary to increase the number of structural sub-steps).